src/services/players/player_service.cpp

```diff
@@ -59,6 +61,69 @@ namespace big
 		return nullptr;
 	}
 
+		player_ptr player_service::get_by_name(std::string_view name) const
+	{
+		std::string self_name  = g_player_service->get_self()->get_name();
+		std::string name_lower = name.data();
+		std::transform(self_name.begin(), self_name.end(), self_name.begin(), ::tolower);
+		std::transform(name_lower.begin(), name_lower.end(), name_lower.begin(), ::tolower);
+
+		for (auto& [_, player] : m_players)
```

Comment:
If you're not making use of the key then consider reworking the for loop to make use of `std::ranges::views::values`.

```cpp
for (const auto& session_type_string : m_session_types | std::ranges::views::values)
{

}
```

Reply:
Unused variable `_`.